attribute-masking-service/src/main/java/uk/gov/gchq/palisade/service/attributemask/stream/config/AkkaRunnableGraph.java

```diff
@@ -69,48 +74,55 @@ KafkaProducerService kafkaProducerService(
-                .mapAsync(PARALLELISM, (Pair<CommittableMessage<String, AttributeMaskingRequest>, String> messageAndToken) -> {
-                    AttributeMaskingRequest request = messageAndToken.first().record().value();
-                    return service.storeAuthorisedRequest(messageAndToken.second(), request)
-                            .thenApply(ignored -> new Tuple3<>(messageAndToken.first(), messageAndToken.second(), request));
-                })
-
-                // Mask resource attributes, keeping track of original message and token
-                .map(messageTokenRequest -> new Tuple3<>(messageTokenRequest.t1(), messageTokenRequest.t2(), service.maskResourceAttributes(messageTokenRequest.t3())))
-
-                // Build producer record, copying the partition, keeping track of original message
-                .map((Tuple3<CommittableMessage<String, AttributeMaskingRequest>, String, AttributeMaskingResponse> messageTokenResponse) -> {
-                    ConsumerRecord<String, AttributeMaskingRequest> requestRecord = messageTokenResponse.t1().record();
-                    // In the future, consider recalculating the token according to number of upstream/downstream partitions available
-                    return new Pair<>(messageTokenResponse.t1(), new ProducerRecord<>(outputTopic.getName(), requestRecord.partition(), requestRecord.key(), messageTokenResponse.t3(), requestRecord.headers()));
+                .mapAsync(PARALLELISM, (Pair<CommittableMessage<String, AttributeMaskingRequest>, String> messageAndToken) ->
+                    service.storeAuthorisedRequest(messageAndToken.second(), messageAndToken.first().record().value())
+                            .thenApply(auditable -> Pair.create(messageAndToken.first(), auditable))
+                )
+
+                // Mask resource attributes, keeping track of original message
+                .map(request -> Pair.create(request.first(), service.maskResourceAttributes(request.second().getAttributeMaskingRequest())
+                        .chain(request.second().getAuditErrorMessage())))
+
+                // Build producer message, copying the partition, keeping track of original message
+                .map((Pair<CommittableMessage<String, AttributeMaskingRequest>, AuditableAttributeMaskingResponse> response) -> {
+                    ConsumerRecord<String, AttributeMaskingRequest> requestRecord = response.first().record();
+                    return (Optional.ofNullable(response.second().getAuditErrorMessage()).isPresent()) ?
+                        ProducerMessage.single(
+                            new ProducerRecord<>(errorTopic.getName(), requestRecord.partition(), requestRecord.key(),
+                                    SerDesConfig.errorValueSerializer().serialize(null, response.second().getAuditErrorMessage()), requestRecord.headers()),
+                                (Committable) response.first().committableOffset())
+                        :
```

Comment:
SonarQube usually throws a smell around the use of the ternary operator, if we are accepting it over an if else, can you add some inline javadoc along the lines of "if theres no error message then produce the message with the attributeMaskingResponse"